src/base/Literal.ml

```diff
@@ -79,6 +79,8 @@ module type ScillaLiteral = sig
 
     val equal : t -> t -> bool
 
+    val sub : t -> pos:int -> len:int -> t
```

Comment:
Could you please comment why you need `sub` in the `Literal` interface?

Reply:
I'm using this in Builtins.ml, to support `substr` for byte strings

Reply:
Ah, makes sense, thanks. I got confused by the diff a bit :)